Plugin~/WebRTCPlugin/UnityVideoTrackSource.cpp

```diff
@@ -20,11 +20,12 @@ namespace webrtc
         : AdaptedVideoTrackSource(/*required_alignment=*/1)
         , is_screencast_(is_screencast)
         , frame_(nullptr)
+        , syncApplicationFramerate_(false)
     {
         taskQueue_ = std::make_unique<rtc::TaskQueue>(
             taskQueueFactory->CreateTaskQueue("VideoFrameScheduler", TaskQueueFactory::Priority::NORMAL));
         scheduler_ = std::make_unique<VideoFrameScheduler>(taskQueue_->Get());
-        scheduler_->Start(std::bind(&UnityVideoTrackSource::CaptureNextFrame, this));
+        scheduler_->Start(std::bind(&UnityVideoTrackSource::OnUpdateVideoFrame, this));
```

Comment:
Shouldn't this line be wrapped with `if (!syncApplicationFramerate_)`?

Or alternatively, `Start` unconditionally but immediately `Pause` if `syncApplicationFramerate_` is `true`? Since I see you're kind of relying on scheduler to be active, in the `SetSyncApplicationFramerate` function below.

Reply:
@vl4dimir
OK, I fixed the constructor to pause the scheduler at first and make `syncApplicationFramerate_` true by default.